src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = newNode;
+            tail = newNode;
+        } else if (size == 1) {
+            Node<T> newNode = new Node<>(null, value, head);
+            head.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> newNode = new Node<>(null, value, tail);
+            newNode.prev.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == 0) {
+            add(value);
+        }
+        indexCheck(index);
+        if (index == 0) {
+            Node<T> newNode = new Node<>(head, value, null);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+
+        } else if (index < size) {
+            Node<T> nodeOnIndex = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(nodeOnIndex, value, nodeOnIndex.prev);
+            nodeOnIndex.prev.next = newNode;
+            nodeOnIndex.prev = newNode;
+            size++;
+        } else {
+            add(value);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
```

Comment:
You can remove this line, because you already check the index in the `getNodeByIndex` method